news_events/etl/ol_events.py

```diff
@@ -5,171 +5,192 @@
-    """
-    url = urljoin(OL_EVENTS_BASE_URL, event_data.find("a").attrs["href"])
-    return get_soup(url) if url else None
+    return [
+        relation.get("attributes", {}).get("name")
+        for relation in relation_data.get("data", [])
+    ]
 
 
-def transform_event_content(event_page_data: Soup) -> str:
+def extract_event_image(image_data: dict) -> tuple[dict, dict]:
     """
-    Get the text for the Open Learning Event element.
+    Extract the image data from the Open Learning Event.
 
     Args:
-        event_page_data (Soup): The event element BeautifulSoup object
+        image_data (dict): The image data
 
     Returns:
-        str: The event element text
-
-    """
-    return tag_text(event_page_data.find("div", class_="field--name-body"))
+        tuple[dict, dict]: The image text metadata and image source metadata
+
+    """
+    img_text_url = image_data.get("links", {}).get("related", {}).get("href")
+    if not img_text_url:
+        return {}, {}
+    img_text_metadata = get_request_json(img_text_url)
+    next_url = (
+        img_text_metadata.get("data", {})
+        .get("relationships", {})
+        .get("field_media_image", {})
+        .get("links", {})
+        .get("related", {})
+        .get("href")
+    )
+    if not next_url:
+        return img_text_metadata, {}
+    img_src_metadata = get_request_json(next_url) if next_url else {}
+    return img_text_metadata, img_src_metadata
 
 
-def transform_event_image(event_data: Soup) -> dict:
+def transform_event_image(image_text_metadata: dict, image_src_metadata: dict) -> dict:
     """
-    Get the image url for the Open Learning Event.
+    Get the image url for the Open Learning Event image.
 
     Args:
-        event_data (Soup): The event data BeautifulSoup object
+        image_text_metadata (dict): The image metadata containing alt/title info
+        image_src_metadata(dict): The image metadata containing the image url
 
     Returns:
-        dict: The event image url
-    """
-    image_div = event_data.find("div", class_="field--name-field-event-image")
-    if image_div:
-        img = image_div.find("img")
-        if img:
-            return {
-                "url": urljoin(OL_EVENTS_BASE_URL, img.attrs["src"]),
-                "alt": img.attrs["alt"],
-                "description": img.attrs["alt"],
-            }
+        dict: The transformed event image info
+    """
+    image_src = (
+        image_src_metadata.get("data", {})
+        .get("attributes", {})
+        .get("uri", {})
+        .get("url")
+    )
+    attrs = (
+        image_text_metadata.get("data", {})
+        .get("relationships", {})
+        .get("field_media_image", {})
+        .get("data", {})
+        .get("meta", {})
+    )
+    if image_src:
+        return {
+            "url": urljoin(OL_EVENTS_BASE_URL, image_src),
+            "alt": attrs.get("alt"),
+            "description": attrs.get("title"),
```

Comment:
curious if there is a specific reason for changing title from the api to description.

Reply:
The front end html for the events site never seems to include a title attribute,  but I noticed it is in the API response.  About half the time it is blank, the other half it is identical to the alt tag, but it seemed worthwhile to record that info here anyway since it was available.